### DISCUSSION TITLE
Helper func for OptNil types

### DISCUSSION THREAD

**Discussion by adrianlungu** [ogen-go/ogen#1225] [Ideas]
Hello,

While using ogen, I've often run into code where we check if an `OptNil` type `IsSet` and `!IsNull` as the end field is a pointer, so I thought, why not have a different helper `Get` function that returns `&Value` or `nil` ?

i.e. if we have 

```
{
   field: "value"
}
or 
{
   field: ""
}
or 
{
   field: null
}
or
{
}
```

If value of `field` is `null` or missing altogether, it would return nil, otherwise it would return a pointer to `""` or `"value"`.

**Comment by ghost** [comment]
`Get` method returns `("", false)` in case if value is absent or is null.

You can use generics to write a simple helper function to get a pointer:
```go
func AsPtr[
	T any,
	OptT interface {
		Get() (T, bool)
	},
](o OptT) *T {
	v, ok := o.Get()
	if !ok {
		return nil
	}
	return &v
}
```

Probably we should add an `AsPtr` method to optionals.

**Reply by adrianlungu** [reply]
Thanks! 

That's what I ended up doing; if I have some time, might make a PR to add them over here as well.
